### PR TITLE
Hyperparameter checks

### PR DESCRIPTION
**Pull request by technoAl**

### CODE REVIEW

**Review by GrantPerkins — CHANGES_REQUESTED** [review]
Can still train with no dataset selected.

File: packages/client/src/components/project/input/Buttons.tsx [hunk 1]
```diff
@@ -30,20 +35,56 @@ const START_TRAINING = gql`
...
 
-  const { data, loading, error } = useQuery<GetDockerState>(GET_DOCKER_STATE, { pollInterval: 5000 });
-  if (loading) return <p>connecting to trainer</p>;
-  if (error) return <p>cant connect to trainer</p>;
-  if (data?.dockerState === DockerState.TRAIN_PULL) return <p>no train image yet</p>;
+  if (
+    (parameters.data?.project?.epochs || 0) <= 0 ||
+    (parameters.data?.project?.batchSize || 0) <= 0 ||
+    (parameters.data?.project?.evalFrequency || 0) <= 0 ||
+    (parameters.data?.project?.percentEval || 0) <= 0
```
Comment (AustinShalit) [inline packages/client/src/components/project/input/Buttons.tsx]: We use this multiple times. Can we extract to a function?

File: packages/client/src/components/project/input/Buttons.tsx [hunk 2]
```diff
@@ -30,20 +35,56 @@ const START_TRAINING = gql`
...
+  ) {
+    return <Button> Invalid Parameters </Button>;
```
Comment (AustinShalit) [inline packages/client/src/components/project/input/Buttons.tsx]: Be sure to set the styling of these buttons to be disabled so users don't try to click them.

File: packages/client/src/components/project/input/Buttons.tsx [hunk 3]
```diff
@@ -30,20 +35,56 @@ const START_TRAINING = gql`
...
+  }
+
+  if (datasetNames.length === 0) {
+    return <Button> Requires Dataset </Button>;
```
Comment (AustinShalit) [inline packages/client/src/components/project/input/Buttons.tsx]: Be sure to set the styling of these buttons to be disabled so users don't try to click them.

**Review by AustinShalit — COMMENTED** [review]
Small feedback.

**Review by GrantPerkins — APPROVED** [review]
Works on my machine :)

**Pull request merged**